classes/local/factor/object_factor_base.php

```diff
@@ -281,7 +281,17 @@ public function revoke_user_factor($factorid = null) {
         global $DB, $USER;
 
         if (!empty($factorid)) {
+            // If we have an explicit factor id, this means we need to be careful about the user.
             $params = ['id' => $factorid];
+            $existing = $DB->get_record('tool_mfa', $params);
+            if (empty($existing)) {
+                return false;
+            }
+            $matchinguser = $existing->userid == $USER->id;
+            if (!is_siteadmin() && !$matchinguser) {
```

Comment:
is there any chance we can add tests to cover new changes?

Reply:
There currently isnt coverage of the abstract class implementation, which is why I haven't already, as its a larger piece to embark on, but I will get that ball rolling now